### PR TITLE
BUGFIX: Cast identifiers to string when calling frontent / backend methods

### PR DESCRIPTION
**Pull request by daniellienert**
Fixes: #1878

### CODE REVIEW

**Comment by daniellienert** [comment]
If this prevents PHP from doing magic, I also prefer this one.

**Comment by kdambekalns** [comment]
> If this prevents PHP from doing magic

No, it undoes the magic. :)

**Comment by daniellienert** [comment]
@kdambekalns you are a Un-Magician :)
Works, added two tests to prove it.

File: Neos.Cache/Classes/Backend/TransientMemoryBackend.php [hunk 1]
```diff
@@ -142,7 +147,7 @@ public function flushByTag(string $tag): int
     {
         $identifiers = $this->findIdentifiersByTag($tag);
         foreach ($identifiers as $identifier) {
-            $this->remove($identifier);
+            $this->remove((string)$identifier);
```
Comment (kdambekalns) [inline Neos.Cache/Classes/Backend/TransientMemoryBackend.php]:
```suggestion
            $this->remove($identifier);
```

Comment (daniellienert) [reply Neos.Cache/Classes/Backend/TransientMemoryBackend.php]: absolutely

**Comment by kitsunet** [comment]
Definitely better than before but I guess a “0123” would come back as “123” still?

**Comment by kdambekalns** [comment]
> Definitely better than before but I guess a “0123” would come back as “123” still?

No. The leading zero makes it work:

```
php > var_dump(['123'=>'foo']);
array(1) {
  [123] => string(3) "foo"
}

php > var_dump(['0123'=>'foo']);
array(1) {
  '0123' => string(3) "foo"
}
```

Thankfully it does not cast it as octal like this:

```
php > var_dump([0123=>'foo']);
array(1) {
  [83] => string(3) "foo"
}
```

**Comment by kitsunet** [comment]
Thanks, weird PHP is weird, but good that it works :)

**Comment by albe** [comment]
Yeah, that's what I checked first :)

**Pull request merged**